drivers/hmis/app/graphql/types/hmis_schema/service_type_input.rb

```diff
@@ -8,14 +8,26 @@ module Types
   class HmisSchema::ServiceTypeInput < BaseInputObject
     description 'Create service type input'
 
-    argument :name, String, required: true
-    argument :service_category_id, ID, required: true
+    argument :name, String, required: false
+    argument :service_category_id, ID, required: false
+    argument :service_category_name, String, required: false
     argument :supports_bulk_assignment, Boolean, required: false
 
     def to_params
-      result = to_h.except(:service_category_id)
-      result[:custom_service_category_id] = service_category_id
-      result
+      to_h.except(:service_category_id, :service_category_name)
+    end
+
+    def get_or_create_service_category(user_id, data_source_id)
```

Comment:
I would avoid using "create" in the method name here, because the method doesn't actually `create` (persist) the custom service category. it just builds it an returns it.  Maybe `find_or_initialize_service_category`, which mimics the language of the Rails built-in find_or_initialize_by.